src/rookify/modules/__init__.py

```diff
@@ -30,7 +30,12 @@ def _load_module(machine: Machine, config: Dict[str, Any], module_name: str) ->
     :return: returns tuple of preflight_modules, modules
     """
```

Comment:
I think `_load_module` does not return anything and maybe parameters could be added:

```suggestion
"""
    Dynamically loads a module from the 'rookify.modules' package and registers its states.

    :param machine: The state machine where the module's states will be registered.
    :param config: Configuration dictionary passed to the module's state registration.
    :param module_name: The name of the module to load. Can be an absolute or relative path.
    
    :raises ModuleLoadException: If the module structure is invalid (missing `ModuleHandler` or `register_states`).
    """
```